frontend/src/metabase/hoc/withReducer.jsx

```diff
@@ -0,0 +1,16 @@
+import React, { useContext } from "react";
+import { StoreContext } from "../store";
+
+export const withReducer = (key, reducer) => Component => {
```

Comment:
Would it be better to have a plugin for reducer instead? E.g. `PLUGIN_REDUCERS = []`, which is by default `[(state) => reducer(state)]`, and the plugin functions are reduced to get a final reducer? This approach is used in multiple places in the app.